torch/csrc/jit/graph_executor.cpp

```diff
@@ -732,13 +733,60 @@ struct GraphExecutorImpl {
   std::mutex compile_mutex;
 };
 
-GraphExecutor::GraphExecutor(std::shared_ptr<Graph> graph, bool optimize)
-    : pImpl(new GraphExecutorImpl(std::move(graph), optimize)) {}
+struct ProfilingGraphExecutorImpl : public GraphExecutorImpl {
+
+  using GraphExecutorImpl::GraphExecutorImpl;
+
+  void run(Stack& stack) override {
+
+    AT_CHECK(
+        stack.size() >= num_inputs,
+        "expected ",
+        num_inputs,
+        " inputs, but got only ",
+        stack.size());
+
+    {
```

Comment:
Can you format your code before PR review? I've already seen one place in the code where this formatting has leaked into the actual repo!